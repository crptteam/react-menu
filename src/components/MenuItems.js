import React from 'react';
import PropTypes from 'prop-types';

import MenuItemsWrapper from '../styled/MenuItemsWrapper';
import MenuItemWrapper from '../styled/MenuItemWrapper';

const propTypes = {
  isOpen: PropTypes.bool.isRequired,
  content: PropTypes.arrayOf(PropTypes.func).isRequired,
  onClick: PropTypes.func.isRequired,
};

const MenuItems = ({ isOpen, content, onClick }) => {
  return (
    <MenuItemsWrapper
      onClick={onClick}
    >
      {content.map((Item, i) => (
        <MenuItemWrapper key={i} isOpen={isOpen}>
          <Item isOpen={isOpen} />
        </MenuItemWrapper>
      ))}
    </MenuItemsWrapper>
  );
};

MenuItems.displayName = 'MenuItems';
MenuItems.propTypes = propTypes;

export default MenuItems;
