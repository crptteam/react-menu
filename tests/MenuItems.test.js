import React from 'react';
import { shallow } from 'enzyme';

import MenuItems from '../src/components/MenuItems';

describe('MenuItems', () => {
  it('should renders without problem', () => {
    const items = [{
      component: jest.fn(),
      isSelected: false,
    }, {
      component: jest.fn(),
      isSelected: true,
    }, {
      component: jest.fn(),
      isSelected: false,
    }];
    const component = shallow(<MenuItems isFull content={items} />);
    expect(component).toMatchSnapshot();
  });

  it('should render every component from items in wrapper', () => {
    const items = [{
      component: jest.fn(),
      isSelected: false,
    }, {
      component: jest.fn(),
      isSelected: true,
    }, {
      component: jest.fn(),
      isSelected: false,
    }];
    const component = shallow(<MenuItems isFull content={items} />);
    expect(
      component
        .find('mockConstructor')
        .length,
    ).toEqual(items.length);

    expect(
      component
        .find('WithTheme(MenuItemWrapper)')
        .length,
    ).toEqual(items.length);
  });

  it('should render component from items with isFull value', () => {
    const items = [{
      component: jest.fn(),
      isSelected: false,
    }];
    const component = shallow(<MenuItems isFull content={items} />);
    expect(
      component
        .find('mockConstructor')
        .at(0)
        .prop('isFull'),
    ).toEqual(true);

    component.setProps({ isFull: false, content: items });
    expect(
      component
        .find('mockConstructor')
        .at(0)
        .prop('isFull'),
    ).toEqual(false);
  });

  it('should render MenuItemWrapper with isFull value', () => {
    const items = [{
      component: jest.fn(),
      isSelected: false,
    }];
    const component = shallow(<MenuItems isFull content={items} />);
    expect(
      component
        .find('WithTheme(MenuItemWrapper)')
        .at(0)
        .prop('isFull'),
    ).toEqual(true);

    component.setProps({ isFull: false, content: items });
    expect(
      component
        .find('WithTheme(MenuItemWrapper)')
        .at(0)
        .prop('isFull'),
    ).toEqual(false);
  });


  it('should render MenuItemWrapper with isSelected value', () => {
    const items = [{
      component: jest.fn(),
      isSelected: false,
    }];
    const component = shallow(<MenuItems isFull content={items} />);
    expect(
      component
        .find('WithTheme(MenuItemWrapper)')
        .at(0)
        .prop('isSelected'),
    ).toEqual(false);

    items[0].isSelected = true;
    component.setProps({ isFull: false, content: items });
    expect(
      component
        .find('WithTheme(MenuItemWrapper)')
        .at(0)
        .prop('isSelected'),
    ).toEqual(true);
  });
});
