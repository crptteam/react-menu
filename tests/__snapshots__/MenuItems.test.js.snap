// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItems should renders without problem 1`] = `
<MenuItemsWrapper>
  <WithTheme(MenuItemWrapper)
    isFull={true}
    isSelected={false}
    key="0"
  >
    <mockConstructor
      isFull={true}
    />
  </WithTheme(MenuItemWrapper)>
  <WithTheme(MenuItemWrapper)
    isFull={true}
    isSelected={true}
    key="1"
  >
    <mockConstructor
      isFull={true}
    />
  </WithTheme(MenuItemWrapper)>
  <WithTheme(MenuItemWrapper)
    isFull={true}
    isSelected={false}
    key="2"
  >
    <mockConstructor
      isFull={true}
    />
  </WithTheme(MenuItemWrapper)>
</MenuItemsWrapper>
`;
